traversal/selector/helpers.go

```diff
@@ -0,0 +1,35 @@
+package selector
+
+import (
+	"strings"
+
+	"github.com/ipld/go-ipld-prime/codec/json"
+	"github.com/ipld/go-ipld-prime/node/basic"
+)
+
+// REVIEW: I feel sketchy about CompileJSONSelector because it brings in json as a transitive dependency.
+// This has consequences as far as dragging go-cid in because our json shares code with dagjson...
+// and then go-cid ends up in the transitives of the traversal package, since traversal depends on selector.
+// That's... not great.
+// On the other hand: trying to keep up that paper wall is creating a lot of strife.  Maybe it's time to give up.
```

Comment:
Big question here about what we let this do to our dependency tree.

I think the usability argument probably needs to win the match here, but it's a pretty major boundary relaxation, so flagging it for some discussion would make me more comfortable with it.

Reply:
if you want to put the helper functions in a separate folder/module to avoid it, that seems okay?

Reply:
They're pretty entangled, not much room for extracting a shared helper. An alternative way might be to switch the dependency so dagjson is an extension of json that adds bytes and cid.

.. or we address the go-cid dependency issues.

Reply:
@willscott has just brought up elsewhere that if we add features like "list of CIDs to stop at" into Selectors, then any codec that can parse that is going to have a go-cid dependency anyway.

So, a separate package here is still an option, and address go-cid's dependency sprawl is still a long term wish, but "fix json to not depend on go-cid" now seems irrelevant because it looks like selectors may end up using dag-json anyway.